### PR TITLE
Extensibility

### PR DESCRIPTION
**Pull request by krlmlr**
# Summary

Enables use of other recorder functions, e.g. `jointprof::comingle_pprof()`.
This is a draft for discussion; e.g. we might want to extend `record_pprof()` to support `...` to pass on to `Rprof()`.

# Checklist

- [x] I understand and agree to `proffer`'s [code of conduct](https://github.com/r-prof/proffer/blob/master/CODE_OF_CONDUCT.md).
- [x] I have listed any substantial changes in the [development news](https://github.com/r-prof/proffer/blob/master/NEWS.md).
- [ ] I have added [`testthat`](https://github.com/r-lib/testthat) unit tests to [`tests/testthat`](https://github.com/r-prof/proffer/tree/master/tests/testthat) for any new functionality. Feel free to use `skip("interactive only")` for any tests that cannot be fully automated.
- [x] This pull request is not a [draft](https://github.blog/2019-02-14-introducing-draft-pull-requests).

### CODE REVIEW

**Comment by wlandau** [comment]
Great idea! I appreciate the convenience of

```r
pprof(expr, recorder = jointprof::comingle_pprof)
```

or

```r
default::default(pprof) <- list(recorder = jointprof::comingle_pprof)
pprof(expr1)
pprof(expr2)
```

so we do not need to manage paths.

```r
path <- jointprof::comingle_pprof(expr)
serve_pprof(path)
```

File: R/pprof.R [hunk 1]
```diff
@@ -18,9 +22,11 @@ pprof <- function(
   host = "localhost",
   port = NULL,
   browse = interactive(),
-  verbose = TRUE
+  verbose = TRUE,
+  recorder = record_pprof,
```
Comment (wlandau) [inline R/pprof.R]: Should we allow a global option?

```r
pprof <- function(
   # args...
  recorder = getOption("pprof_recorder"),
  ...
) {
  recorder <- recorder %||% record_pprof
  # ...
}
```

**Review by wlandau — CHANGES_REQUESTED** [review]

File: R/pprof.R [hunk 2]
```diff
@@ -6,6 +6,10 @@
 #'   to take down the server with `$kill()`.
 #' @inheritParams serve_pprof
 #' @param expr R code to run and profile.
+#' @param recorder A function that records a `pprof` profile
+#'   for an expression. Default: [record_pprof].
```
Comment (wlandau) [inline R/pprof.R]: Maybe `jointprof::comingle_pprof()` deserves an explicit mention so users understand why this argument exists.

Comment (krlmlr) [reply R/pprof.R]: Sure. For now we need to point to the documentation on the pkgdown website.

File: R/pprof.R (hunk 1 above)
Comment (krlmlr) [reply R/pprof.R]: We could also implement a wrapper in {jointprof}.

**Comment by krlmlr** [comment]
I have implemented `joint_pprof()` in {jointprof}, removing the need for the `recorder` argument. I still think the dots are useful.

What should the tests look like?

**Comment by wlandau** [comment]
Thanks! I suppose `...` will only have arguments to `Rprof()` now. Maybe we could test the `append` argument and see if the output file from `Rprof()` increases in size? Do you want to handle that in this PR or a different one?

**Comment by krlmlr** [comment]
Added test. Profiles created with `append = TRUE` currently trigger a warning in {profile}.

**Review by wlandau — APPROVED** [review]

**Comment by wlandau** [comment]
Looks great, thanks!

**Pull request merged**